dts/bindings/gpio/cypress,psoc6-gpio.yaml

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) 2020 ATL Electronics
+# SPDX-License-Identifier: Apache-2.0
+
+description: Cypress GPIO PORT node
+
+compatible: "cypress,psoc6-gpio"
+
+include: [gpio-controller.yaml, base.yaml]
+
+properties:
+    reg:
+      required: true
+
+    interrupts:
+      required: true
+
+    label:
+      required: true
+
+    "#gpio-cells":
+      const: 2
+
+    "#cypress,pin-cells":
+      type: int
+      required: true
+      const: 4
+      description: Number of items to expect in a cypress,pins specifier
+
+gpio-cells:
+  - pin
+  - flags
+
+cypress,pin-cells:
+  - port
+  - pin
+  - peripheral
+  - flags
```

Comment:
we should handle flags like we are doing on atmel as explicit properties

Reply:
Done!

```
#define DT_CYPRESS_HSIOM_FLAG(flag) flag;
#define DT_CYPRESS_HSIOM_FLAGS(...) \
	MACRO_MAP_CAT(DT_CYPRESS_HSIOM_FLAG __VA_OPT__(,) __VA_ARGS__)

#define DT_CYPRESS_HSIOM(inst, signal, port, pin, hsiom, ...) \
	p##port##_##pin##_##inst##_##signal: \
	p##port##_##pin##_##inst##_##signal { \
	cypress,pins = < &p##port port pin HSIOM_SEL_##hsiom > ; \
		DT_CYPRESS_HSIOM_FLAGS(__VA_ARGS__) \
	}
```
Now it have a variable length flags list. For an entry like this `DT_CYPRESS_PIN(uart5, rx, 5, 0, act_4, bias-pull-up, input-buffer);` It will construct as:
```
			p5_0_uart5_rx: p5_0_uart5_rx {
				cypress,pins = < &p5 0x5 0x0 0x12 >;
				bias-pull-up;
				input-buffer;
			};
```
The flags list can be zero length.

Drawback was, I needed to copy this https://github.com/zephyrproject-rtos/zephyr/pull/28645/commits/1f0c490272f107f8c8f6324d9b5812ff99d7bb9f#diff-e9536070e740fa654129a980ece66553R89-R98 from <sys/util.h> and add <sys/util_internal.h> to make it works. When include <sys/util.h> I got many path errors and garbage at **./build/zephyr/\<board\>.dts.pre.tmp**.

I send an update to show the whole solution and would like know how solve the copied content thing. Since <sys/util_internal.h> I should not use it (I think) and the content copied can not be on the <sys/util_internal.h> I guess.

Reply:
Proposal fix 0eac735dc23cda700d53984bffd32894b18ab3d9